transports/vm/src/main/java/org/mule/transport/vm/VMMessageReceiver.java

```diff
@@ -197,21 +201,24 @@ public MuleEvent process() throws Exception
         int batchSize = getBatchSize(queue.size());
 
         // try to get the first event off the queue
-        MuleEvent message = (MuleEvent) queue.poll(connector.getQueueTimeout());
+        MuleMessage message = (MuleMessage) queue.poll(connector.getQueueTimeout());
 
         if (message != null)
         {
-            // keep first dequeued event
-            ((DefaultMuleMessage)message.getMessage()).setMuleContext(endpoint.getMuleContext());
-            messages.add(message.getMessage());
+            // keep first dequeued message
+            if (message instanceof DefaultMuleMessage)
+            {
+                ((DefaultMuleMessage) message).setMuleContext(endpoint.getMuleContext());
+            }
+            messages.add(message);
 
             // keep batching if more events are available
             for (int i = 0; i < batchSize && message != null; i++)
             {
-                message = (MuleEvent)queue.poll(0);
+                message = (MuleMessage) queue.poll(0);
```

Comment:
This poll assumes no events are left in a persistent queue when upgrading to this version.